todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/taskdetail/TaskDetailFragment.java

```diff
@@ -190,18 +199,33 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
 
     @Override
     public void showTitle(@NonNull String title) {
-        mDetailTitle.setVisibility(View.VISIBLE);
-        mDetailTitle.setText(title);
+        getDetailTitle().setVisibility(View.VISIBLE);
+        getDetailTitle().setText(title);
     }
 
     @Override
     public void showMissingTask() {
-        mDetailTitle.setText("");
-        mDetailDescription.setText(getString(R.string.no_data));
+        getDetailTitle().setText("");
+        getDetailDescription().setText(getString(R.string.no_data));
     }
 
     @Override
     public boolean isActive() {
         return isAdded();
     }
+
+    @NonNull
```

Comment:
This seems complicated just to manage nulls... the app is going to crash anyway if the presenter is null in any of the other methods. Annotating every single field seems excesive. Do you think it's worth it?